src/Sidekick.Business/Apis/Poe/Parser/ParserService.cs

```diff
@@ -73,80 +81,95 @@ public async Task<ParsedItem> ParseItem(string itemText)
             }
         }
 
-        private void ParseHeader(ref ParsedItem item, ref string input)
+        private ParsedItem ParseItemDetails(string itemText, ItemSections itemSections, ItemData itemData)
         {
-            var dataItem = itemDataService.ParseItem(input);
-
-            item.Name = dataItem.Name;
-            item.TypeLine = dataItem.Type;
-            item.Rarity = dataItem.Rarity;
-
-            if (item.Rarity == Rarity.Unknown)
+            var item = new ParsedItem
             {
-                var lines = NewlinePattern.Split(input);
-                item.Rarity = GetRarity(lines[0]);
-            }
+                ItemText = itemText,
+                Name = itemData.Name,
+                TypeLine = itemData.Type,
+                Rarity = itemData.Rarity
+            };
 
-            if (string.IsNullOrEmpty(item.Name) && string.IsNullOrEmpty(item.TypeLine))
+            switch (item.Rarity)
             {
-                throw new NotSupportedException("Item not found.");
+                case Rarity.DivinationCard:
+                case Rarity.Currency:
+                case Rarity.Prophecy:
+                    break;
+
+                case Rarity.Gem:
+                    ParseGem(item, itemSections);
+                    break;
+
+                case var _ when ItemIsMap(itemSections):
+                    ParseMap(item, itemSections);
+                    ParseMods(item);
+                    break;
+
+                default:
+                    ParseEquipmentProperties(item, itemSections);
+                    ParseMods(item);
+                    ParseSockets(item);
+                    ParseInfluences(item, itemSections);
+                    break;
             }
 
-            if (item.Rarity != Rarity.DivinationCard)
-            {
-                item.ItemLevel = GetInt(patterns.ItemLevel, input);
-                item.Identified = !patterns.Unidentified.IsMatch(input);
-                item.Corrupted = patterns.Corrupted.IsMatch(input);
-            }
+            return item;
         }
 
-        private Rarity GetRarity(string rarityString)
+        private bool ItemIsMap(ItemSections itemTextBlock)
         {
-            foreach (var pattern in patterns.Rarity)
-            {
-                if (pattern.Value.IsMatch(rarityString))
-                {
-                    return pattern.Key;
-                }
-            }
-            throw new Exception("Can't parse rarity.");
+            return itemTextBlock.TryGetMapTierLine(out var mapTierLine) && patterns.MapTier.IsMatch(mapTierLine);
```

Comment:
Instead of creating logic here to identify the type of item. I suggest modifying the ParseItem method of the ItemDataService. I would add a property of a new enum that would identify the type (Map, Gem, Prophecy, etc.). It would be similar to the Rarity enum but could also contain Armour, Weapon, etc. That could be identified by modifying the init method.

Reply:
Will work on this 🙂

Reply:
This has been implemented. We can work on fine tuning the parsing for specific categories in the future.